File: in_Python/0788 Rotated Digits.py
class Solution(object):
    def rotatedDigits(self, N):
        """
        :type N: int
        :rtype: int
        """
        res = 0
        for i in range(1,N+1):
            backup = i
            cur = 0
            p = 0
            flag = True
            while i != 0:
                x = i % 10
                if x in [3,4,7]:
                    flag = False
                    break
                if x == 2:
                    x = 5
                elif x == 5:
                    x = 2
                elif x == 6:
                    x = 9
                elif x == 9:
                    x = 6
                else:
                    x = x
                cur += x * (10 **p)
                p += 1
                i /= 10
            if not flag: continue
            if cur != backup: 
                res += 1
        return res

File: in_Java/0788 Rotated Digits.java
class Solution {
    public int rotatedDigits(int N) {
        HashMap<Integer,Integer> hm = new HashMap<>();
        hm.put(1,1);hm.put(0,0); hm.put(8,8); hm.put(2,5);
        hm.put(5,2);hm.put(6,9); hm.put(9,6);
        int res = 0;
        boolean able = true;
        for(int i = 1; i <= N; i++){
            int back = i;
            int cur = 0;
            int power = 0;
            while(i != 0){
                int t = i % 10;
                i /= 10;
                if(!hm.containsKey(t)){
                    able = false;
                    break;
                }else{
                    cur += hm.get(t) * Math.pow(10,power);
                    able = true;
                    power++;
                }
            }
            if(able && cur != back) res++;
            i = back;
        }
        return res;
    }
}